BasicSample/Assets/GrabAndThrow/Scripts/GrabAndThrowBehavior.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+using UnityEngine.XR.OpenXR.Features.Interactions;
+
+public class GrabAndThrowBehavior : MonoBehaviour
+{
+    private bool[] m_wasTriggerPressed = { false, false };
+
+    // Update is called once per frame
+    private void Update()
+    {
+        // Get trigger and velocity data from the left device
+        for (int i = 0; i < 2; i++)
+        {
+            InputDevice device = InputDevices.GetDeviceAtXRNode((i == 0) ? XRNode.RightHand : XRNode.LeftHand);
+            bool deviceHasData = device.TryGetFeatureValue(CommonUsages.isTracked, out bool deviceIsTracked);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.primaryButton, out bool isdeviceTapped);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceVelocity, out Vector3 deviceVelocity);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceAngularVelocity, out Vector3 deviceAngularVelocity);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 devicePosition);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceRotation, out Quaternion deviceRotation);
+
+            if (deviceHasData && deviceIsTracked)
+            {
+                if (m_wasTriggerPressed[i] && !isdeviceTapped)
+                {
+                    // Release the rigidbody using velocity
+                    // Note that Unity's velocity is not expected to match the input's velocity, this is yet to be addressed 
+                    GetComponent<Rigidbody>().velocity = deviceVelocity;
+                    GetComponent<Rigidbody>().angularVelocity = deviceAngularVelocity;
+                    GetComponent<Rigidbody>().isKinematic = false;
+                    m_wasTriggerPressed[i] = false;
+                }
+
+                if (isdeviceTapped)
+                {
+                    // Pick the rigidbody
```

Comment:
For "pick" and throw the object:
1. Decide pickup: When "select is change to pressed": detect if hand pose is close to the object, should it pick up the object or not.  using device position is inside the object, likely Bounds.Contains,  and add a bit buffer of half of hand size (5cm) (e.g. Unity's Physics.CheckSphere,  pick up if the sphere at devicePose intercept the object).
2. Remember the offset pose from the device pose to the object pose.
3. Move the object with hand:  On each Update, if the object is picked up, update the object transform using the above offset + device pose.
4. Decide to throw:  when "select is changed to release":  a) get the pose of centerOfMass, b) get the throw offset = pickup_offset + centerOfMass, c) compute the velocity on throw_offset from the velocity of devicePose. (apply the lever arm math).  set the velocity to the object.   Then set "use gravity" to true.